### ISSUE TITLE
Accessibility: Inconsistent focus outlines across the CMS

### ISSUE THREAD

**Issue by GuySartorelli** [silverstripe/.github#443]
[WCAG 2.2 AA F78](https://www.w3.org/TR/WCAG20-TECHS/F78.html) 2.4.7

Focus outlines (`:focus-visible`) across the CMS are inconsistent and in some cases violate accessible contrast standards.

## Elements with insufficient contrast or inconsistent focus outlines

This is not an exhaustive list - check everything you can think of

- [x] Locales dropdown selector in left menu for `tractorcow/silverstripe-fluent`
- [x] "Add new page" button in `/admin/pages` (and by implied extension, probably all buttons)
- [x] The batch action button and related elements in `/admin/pages`
- [x] Create, Cancel, save, publish (etc) buttons
- [x] Gridfield header buttons (export, import, print, search/filter)
- [x] Gridfield action buttons (edit, view, delete, `...`, etc in rows)
- [x] "Upload", "add folder", "Search", and "list view" buttons in asset admin
- [x] Sort dropdown in asset admin (completely missing focus outline)
- [x] Individual links in linkfield
- [x] draggable element for files and folders in asset admin
- [x] buttons vs field input vs links focus outlines differ
- [x] Left menu - most items only have a top-border visible for focus, not a full outline
- [x] Sort headers in gridfield - only has bottom-border visible for focus, not a full outline (which is also what's used to denote the current sort)
- [ ] ~~Sort headers in asset admin table view - no visible focus state at all~~ - just aren't in the tab index which isn't part of this card.
- [x] Breadcrumbs in a modeladmin and pages admin - link outlines are hard to see in full
- [x] Archive/delete button in linkfield focus state was noted as "not correct"
- [x] Various elements in the history tab, especially when comparing two versions
- [x] Dropdown for field type in CMS for user forms form fields
- [x] Dropdowns for field name and comparison type when adding custom display rules in userforms
- [ ] ~~Highlighted options in "select upload field" dropdown for file upload editable field in userforms.~~ that item isn't focused - in a dropdown it's the actual select element that's focused not the individual options.
- [x] Buttons in login form (login-forms module) and MFA form (check the whole flow incl reset password)
- [x] input fields in login form (login-forms module)

## Notes

- Seek designer input on what the consistent outline should be - and if some elements _should_ have different outlines.
- Some elements may need a different outline if their background or border colour would not have enough contrast against the focus outline. Seek designer feedback for these elements especially

## Acceptance criteria

- [x] Focus outlines are consistent across the CMS
- [x] Focus outlines are displayed for all interactive elements when they have focus
- [x] Focus outlines meet accessibility contrast standards
- [x] The changelog clearly calls out that developers may want to check their custom CSS matches the new focus states

## PRs

- https://github.com/silverstripe/silverstripe-admin/pull/2089
- https://github.com/silverstripe/silverstripe-versioned-admin/pull/449
- https://github.com/silverstripe/silverstripe-advancedworkflow/pull/616
- https://github.com/silverstripe/silverstripe-elemental/pull/1397
- https://github.com/silverstripe/silverstripe-campaign-admin/pull/386
- https://github.com/silverstripe/silverstripe-mfa/pull/628
- https://github.com/silverstripe/silverstripe-login-forms/pull/232
- https://github.com/silverstripe/silverstripe-tagfield/pull/340
- https://github.com/silverstripe/developer-docs/pull/862

**Comment by cassiezoest** [comment]
I think for most elements, having the same focus state will be okay. I'd suggest going with a full outline in a dark blue colour. 

Whatever default colour we're currently using seems to have good/sufficient contrast with light backgrounds. For dark backgrounds such as the top of the lefthand sidebar which shows the site name/username/logout button, we should use a white outline.

<img width="1323" height="351" alt="Image" src="https://github.com/user-attachments/assets/24bc9f67-3fc0-4a86-b351-c8260c02d681" />

For coloured buttons such as Save/Publish, we will need a bit of an offset so that the outline doesn't touch the button itself. For grey/white buttons an offset probably isn't needed.

**Comment by GuySartorelli** [comment]
Review comments after the above was implemented:

> - These look pretty thick, could we try 1px?
> - For the dark focus, could we use `#2A62BC`

**Comment by GuySartorelli** [comment]
Changes have since been approved (using 1.5px outline width) by Cassie from a UX perspective.

**Comment by emteknetnz** [comment]
WCAG specifies minimum 2px width https://www.w3.org/WAI/WCAG22/Understanding/focus-appearance.html

The 1.5px looks like it renders as 1px

e.g.

<img width="298" height="156" alt="Image" src="https://github.com/user-attachments/assets/741ed945-1c25-4951-8a0f-f59394adc5f3" />

vs how it was before

<img width="301" height="141" alt="Image" src="https://github.com/user-attachments/assets/9d492d43-ac59-4dda-8721-0c57ef23e446" />

**Comment by GuySartorelli** [comment]
> The 1.5px looks like it renders as 1px

I told Cassie it was rendering as 1px for me and she says for her using chrome it is visually distinct from 1px.
Wasn't aware of the >= 2px requirement from WCAG though, good spotting. I'll update to use 2px.

FYI @cassiezoest
